src/CommandCache/CachedAnnotatedCommandFactory.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace Pantheon\Terminus\CommandCache;
+
+use Consolidation\AnnotatedCommand\AnnotatedCommandFactory;
+use Consolidation\AnnotatedCommand\Parser\CommandInfo;
+use Pantheon\Terminus\DataStore\DataStoreAwareInterface;
+use Pantheon\Terminus\DataStore\DataStoreAwareTrait;
+use Symfony\Component\Console\Input\InputOption;
+
+class CachedAnnotatedCommandFactory extends AnnotatedCommandFactory implements DataStoreAwareInterface
+{
+    use DataStoreAwareTrait;
+
+  /**
+   * Register all commands and hooks. Read info from cache if possible.
+   *
+   * @param $commandFileInstance
+   * @param null $includeAllPublicMethods
+   * @return array
+   */
+    public function createCommandsFromClass(
+        $commandFileInstance,
+        $includeAllPublicMethods = null
+    ) {
+
+        // Deprecated: avoid using the $includeAllPublicMethods in favor of the setIncludeAllPublicMethods() accessor.
```

Comment:
No need to copy backwards-compatible code when cloning command factory methods!

No matter; let's consider pushing this up to the upstream to avoid maintenance and code-duplication problems.